onnxruntime/core/providers/webgpu/webgpu_external_header.h

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
```

Comment:
do we no longer need to use macro HAS_* like this https://github.com/microsoft/onnxruntime/blob/655a23ff1d19927d048a325be8065185b127ed14/cmake/CMakeLists.txt#L750

Reply:
We only support GCC 11 and above. "--strict-aliasing" is available in these GCC versions.